gemrb/core/Scriptable/Actor.cpp

```diff
@@ -8184,14 +8184,36 @@ void Actor::AddExperience(int exp, int combat)
 		bonus += adjustmentPercent;
 	}
 	bonus += GetFavoredPenalties();
-	exp = ((exp * (100 + bonus)) / 100) + BaseStats[IE_XP];
+
+	int classid = GetActiveClass() -1; //-1 to allow lookup in xpcap array
```

Comment:
You can check IE_CLASS directly and also only for pst. Plus, please use a consistent naming style. I think the most common one is camel case for vars, which is true at least for this function.

Reply:
The classid var is used at the end of the function to lookup the xp cap from an array, hence the starting at 0 - I didn't change that part, I just moved the initialisation of it upward because I wanted to reuse the value

But it sounds like it isn't needed at all, if I can just use BaseStats[IE_CLASS] in both cases?

Reply:
Actually it looks like it's important to keep this as a call to GetActiveClass, since it checks for dual classing in the other games.

Reply:
You can internalize the +1 to the case values.

Reply:
Right, didn't see you moved it — thought it was new.